src/Extend/DataExchange.py

```diff
@@ -61,319 +67,55 @@ def read_step_file(filename, return_as_shapes=False, verbosity=True):
             failsonly = False
             step_reader.PrintCheckLoad(failsonly, IFSelect_ItemsByEntity)
             step_reader.PrintCheckTransfer(failsonly, IFSelect_ItemsByEntity)
-        transfer_result = step_reader.TransferRoot(1)
-        if not transfer_result:
-            raise AssertionError("Transfer failed.")
+        ok = step_reader.TransferRoot(1)
         _nbs = step_reader.NbShapes()
-        if _nbs != 1:
-            raise AssertionError("Number of shapes is not one.")
         shape_to_return = step_reader.Shape(1)  # a compound
-        if shape_to_return.IsNull():
-            raise AssertionError("Shape is null.")
+        assert not shape_to_return.IsNull()
     else:
         raise AssertionError("Error: can't read file.")
     if return_as_shapes:
-        shape_to_return = TopologyExplorer(shape_to_return).solids()
+        shape_explore = TopologyExplorer(shape_to_return).faces()
+        shape_to_return = []
+        for i in shape_explore:
+            shape_to_return.append(i)
 
     return shape_to_return
 
-
 def write_step_file(a_shape, filename, application_protocol="AP203"):
     """ exports a shape to a STEP file
     a_shape: the topods_shape to export (a compound, a solid etc.)
     filename: the filename
     application protocol: "AP203" or "AP214"
     """
     # a few checks
-    if a_shape.IsNull():
-        raise AssertionError("Shape %s is null." % a_shape)
-    if application_protocol not in ["AP203", "AP214IS"]:
-        raise AssertionError("application_protocol must be either AP203 or AP214IS. You passed %s." % application_protocol)
+    assert not a_shape.IsNull()
+    assert application_protocol in ["AP203", "AP214IS"]
     if os.path.isfile(filename):
         print("Warning: %s file already exists and will be replaced" % filename)
     # creates and initialise the step exporter
     step_writer = STEPControl_Writer()
-    Interface_Static_SetCVal("write.step.schema", application_protocol)
+    Interface_Static_SetCVal("write.step.schema", "AP203")
 
     # transfer shapes and write file
     step_writer.Transfer(a_shape, STEPControl_AsIs)
     status = step_writer.Write(filename)
 
-    if not status == IFSelect_RetDone:
-        raise AssertionError("Error while writing shape to STEP file.")
-    if not os.path.isfile(filename):
-        raise AssertionError("File %s was not saved to filesystem." % filename)
-
-
-def read_step_file_with_names_colors(filename):
```

Comment:
not sure why this method is deleted; it has no overlap with the `pyside2` backend